packages/flutter/test/widgets/heroes_test.dart

```diff
@@ -2255,4 +2274,36 @@ Future<void> main() async {
       moreOrLessEquals(tester.getTopLeft(find.text('1')).dx, epsilon: 0.01)
     );
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/38183.
+  testWidgets('invalidate user gesture driven flights when the gesture is invalid', (WidgetTester tester) async {
```

Comment:
To help connect this test with the code, maybe: "remove user gesture driven flights with the gesture goes stale"